File: two_pointers/naive_two_sum.go
package twopointers

// list has to be sorted
// we go through the list twice hence O(nxn)
func TwoSum(list []int, sum int) bool {
	for i := 0; i < len(list); i++ {
		for j := 0; j < len(list); j++ {

			if i == j {
				continue
			}

			if list[i]+list[j] == sum {
				return true
			}

			if list[i]+list[j] > sum {
				break
			}
		}
	}

	return false
}

File: two_pointers/naive_two_sum.zig
const std = @import("std");

// list has to be sorted -> to be checked
fn NaiveTwoSUm(list: [3]usize, sum: usize) bool {
    for (list) |i| {
        for (list) |j| {
            if (i == j) {
                continue;
            }

            if (list[i] + list[j] == sum) {
                return true;
            }

            if (list[i] + list[j] > sum) {
                break;
            }
        }
    }

    return false;
}

test "naivly test which two numbers add to a given sum" {
    try std.testing.expect(NaiveTwoSUm([3]usize{ 1, 2, 3 }, 5) == true);
}

// checkout slices!
